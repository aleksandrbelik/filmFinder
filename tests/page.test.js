import React from 'react';
import Page from 'Common/page/page';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const tree = renderer
    .create(<Page header={<span>header</span>} headerLine={<span>headerLine</span>} content={<span>content</span>} isFilmDetails />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders correctly without headerLine prop', () => {
  const tree = renderer
    .create(<Page header={<span>header</span>} content={<span>content</span>} isFilmDetails />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders correctly without isFilmDetails prop', () => {
  const tree = renderer
    .create(<Page header={<span>header</span>} headerLine={<span>headerLine</span>} content={<span>content</span>}/>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders correctly without headerLine and isFilmDetails props', () => {
  const tree = renderer
    .create(<Page header={<span>header</span>} content={<span>content</span>}/>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});