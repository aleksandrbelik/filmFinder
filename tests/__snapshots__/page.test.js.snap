// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="page is-film-details"
>
  <div
    className="page__header"
  >
    <div
      className="page__header__container"
    >
      <div
        className="page__header__main"
      >
        <div
          className="page__header__top"
        >
          <div
            className="page__header__name"
          >
            netflixroulette
          </div>
          <button
            className="page__header__button"
            type="button"
          >
            SEARCH
          </button>
        </div>
        <div
          className="page__header__cnt"
        >
          <span>
            header
          </span>
        </div>
      </div>
    </div>
  </div>
  <div
    className="page__header-line"
  >
    <span>
      headerLine
    </span>
  </div>
  <div
    className="page__content"
  >
    <span>
      content
    </span>
  </div>
  <div
    className="page__footer"
  >
    netflixroulette
  </div>
</div>
`;

exports[`renders correctly without headerLine and isFilmDetails props 1`] = `
<div
  className="page"
>
  <div
    className="page__header"
  >
    <div
      className="page__header__container"
    >
      <div
        className="page__header__main"
      >
        <div
          className="page__header__top"
        >
          <div
            className="page__header__name"
          >
            netflixroulette
          </div>
        </div>
        <div
          className="page__header__cnt"
        >
          <span>
            header
          </span>
        </div>
      </div>
    </div>
  </div>
  <div
    className="page__header-line"
  />
  <div
    className="page__content"
  >
    <span>
      content
    </span>
  </div>
  <div
    className="page__footer"
  >
    netflixroulette
  </div>
</div>
`;

exports[`renders correctly without headerLine prop 1`] = `
<div
  className="page is-film-details"
>
  <div
    className="page__header"
  >
    <div
      className="page__header__container"
    >
      <div
        className="page__header__main"
      >
        <div
          className="page__header__top"
        >
          <div
            className="page__header__name"
          >
            netflixroulette
          </div>
          <button
            className="page__header__button"
            type="button"
          >
            SEARCH
          </button>
        </div>
        <div
          className="page__header__cnt"
        >
          <span>
            header
          </span>
        </div>
      </div>
    </div>
  </div>
  <div
    className="page__header-line"
  />
  <div
    className="page__content"
  >
    <span>
      content
    </span>
  </div>
  <div
    className="page__footer"
  >
    netflixroulette
  </div>
</div>
`;

exports[`renders correctly without isFilmDetails prop 1`] = `
<div
  className="page"
>
  <div
    className="page__header"
  >
    <div
      className="page__header__container"
    >
      <div
        className="page__header__main"
      >
        <div
          className="page__header__top"
        >
          <div
            className="page__header__name"
          >
            netflixroulette
          </div>
        </div>
        <div
          className="page__header__cnt"
        >
          <span>
            header
          </span>
        </div>
      </div>
    </div>
  </div>
  <div
    className="page__header-line"
  >
    <span>
      headerLine
    </span>
  </div>
  <div
    className="page__content"
  >
    <span>
      content
    </span>
  </div>
  <div
    className="page__footer"
  >
    netflixroulette
  </div>
</div>
`;
